app/src/main/java/com/pajato/android/gamechat/common/PlayModeManager.java

```diff
@@ -52,32 +64,63 @@
 
     // Public instance variables
 
-    /** Identifies the types of play. */
-    public enum PlayModeType {computer, local, user}
-
     // Protected instance variables
 
-    /** The current play mode popup menu for the experience being enjoyed. */
-    private PopupMenu mPlayModePopup;
-
-    // Private  instance variables
-
-    /** The listener for menu item clicks on the play mode popup menu */
-    private PlayModeClickListener mListener = new PlayModeClickListener();
+    /** The current play mode menu for the experience being enjoyed. */
+    private PopupWindow mPlayModePopupWindow;
 
     // Public instance methods.
 
+    /** Close the play-mode popup window */
+    public void closePlayModeMenu() {
+        mPlayModePopupWindow.dismiss();
+    }
+
     /** Dismiss the play mode menu if it is not null */
     public void dismissPlayModeMenu() {
-        if (mPlayModePopup != null)
-            mPlayModePopup.dismiss();
+        if (mPlayModePopupWindow != null)
+            mPlayModePopupWindow.dismiss();
+    }
+
+    /** Handle a use selection in the play mode menu */
+    public void handlePlayModeUserSelection(View view, BaseExperienceFragment fragment) {
+        Object payload = view.getTag();
+        if (payload == null || !(payload instanceof PlayModeMenuEntry))
+            return;
+        PlayModeMenuEntry entry = (PlayModeMenuEntry) payload;
+        // Handle selecting another User by chaining to the fragment that will select the
+        // User, copy the experience to a new room, and continue the game in that room with
+        // the current state.
+        Account member = MemberManager.instance.getMember(entry.groupKey, entry.accountKey);
+        if (member == null)
+            return;
+        String userName = String.format(Locale.US, "%s (%s)", member.getNickName(), member.email);
+        ListItem selectUserListItem = new ListItem(selectUser, entry.groupKey,
+                entry.accountKey, userName, GroupManager.instance.getGroupName(entry.groupKey),
+                member.url);
+        Experience experience = fragment.getExperience();
+        ListItem expListItem = new ListItem(experience);
+        JoinManager.instance.joinRoom(selectUserListItem);
+
+        List<Player> players = fragment.getExperience().getPlayers();
+        for (Player p : players) {
+            if (p.id == null) {
+                p.id = member.id;
+                p.name = member.getNickName();
+                break;
+            }
+        }
+        fragment.getExperience().setName(fragment.createTwoPlayerName(experience.getPlayers(),
```

Comment:
It might be better to reuse the experience variable than to repeat fragment.getExperience().